app/routes/joblistings/components/JoblistingEditor.js

```diff
@@ -24,195 +27,249 @@ type Props = {
   joblistingId?: string,
   joblisting: Joblisting,
   handleSubmit: Function => void,
-  submitJoblisting: Workplace => void,
-  deleteJoblisting: ID => void,
-  company?: Object,
+  submitJoblisting: Workplace => Promise<*>,
+  deleteJoblisting: ID => Promise<*>,
+  company: SelectInputObject,
   dispatch: any => void,
+  push: string => void,
   isNew: boolean,
-  fetching: boolean
+  fetching: boolean,
+  fetchCompanyContacts: ({ companyId: ID }) => Promise<*>
+};
+
+type State = {
+  responsibleOptions: Array<Object>
+};
+
+type SelectInputObject = {
+  label: string,
+  value: ID
 };
 
-function JoblistingEditor({
-  handleSubmit,
-  joblistingId,
-  joblisting,
-  isNew,
-  submitJoblisting,
-  deleteJoblisting,
-  company,
-  dispatch,
-  fetching = false
-}: Props) {
-  const onSubmit = newJoblisting => {
+class JoblistingEditor extends Component<Props, State> {
+  state = {
+    responsibleOptions: []
+  };
+
+  onSubmit = newJoblisting => {
     const workplaces = newJoblisting.workplaces
       ? newJoblisting.workplaces.map(obj => ({ town: obj.value }))
       : null;
 
-    submitJoblisting({
-      ...newJoblisting,
-      workplaces
+    return this.props
+      .submitJoblisting({
+        ...newJoblisting,
+        id: this.props.joblistingId,
+        applicationUrl: httpCheck(newJoblisting.applicationUrl),
+        workplaces
+      })
+      .then(result => {
+        const id = this.props.joblistingId || result.payload.result;
```

Comment:
is `.result` an ID here?

Reply:
Payload is the "result" after `normalizr`, and `.result` the `idAttribute` from the normalized object.